compiler/resolution/interfaceResolution.cpp

```diff
@@ -2113,16 +2177,25 @@ static ImplementsStmt* findSatisfyingIstm(InterfaceSymbol* isym,
   FirstPick pick = pickMatchingImplementsStmts(isym, visibleFns, call2wf);
   ImplementsStmt* bestIstm = NULL;
 
-  if (ImplementsStmt* conSuccess = pick.conSuccess)
+  if (ImplementsStmt* conSuccess = pick.conSuccess) {
     // yippee, found a satisfying concrete istm
     bestIstm = conSuccess;
-
-  else if (ImplementsStmt* genSuccess = pick.genSuccess)
+  } else if (ImplementsStmt* genSuccess = pick.genSuccess) {
     // instantiate a generic istm
     bestIstm = useGenericImplementsStmt(callsite, isym, call2wf, genSuccess);
-
-  else if (fInferImplementsStmts)
-    bestIstm = checkInferredImplStmt(callsite, isym, call2wf, pick.failure);
+  } else {
+    if (fInferImplementsStmts || isAutoImplementInternalInterface(isym)) {
+      // If we're trying to infer a compiler-generated interface, we only
+      // want to use compiler-generated procedures, since the whole point
+      // of interfaces for library procedures is to avoid user-defined methods
+      // from accidentally being used in a special way. However, if fInferImplementsStmts
+      // is set, the user is specifically opting into such uses. So,
+      // set generatedOnly to !fInferImplementsStmts, which given the condition
+      // of this branch implies that this is a special interface.
```

Comment:
I think perhaps we might be able to simply remove `fInferImplementsStmts`. I don't think we need it.